### PR TITLE
Set filename to '<string>' in run_parser in case tok->filename is NULL

### LINKED ISSUE #166: parse_string segfaults when using mode=2
Due to `tok->filename` being NULL, when `parse_string(..., mode=2)` is called, [Python/compile.c:336](https://github.com/python/cpython/blob/558f07891170fe5173f277d3749e92d844de0a27/Python/compile.c#L336) segfaults. What's the best course of action here?

### PR DESCRIPTION
**Pull request by lysnikolaou**
Fixes #166.

### CODE REVIEW

**Comment by lysnikolaou** [comment]
> This will leak the object created.

Is it okay if I add a `Py_XDECREF` after `PyAST_CompileObject` has been called?

**Comment by gvanrossum** [comment]
> Is it gonna be okay if I add a Py_XDECREF after PyAST_CompileObject has been called?

Sure, but only if PyUnicode_FromString() was called -- otherwise you'd be stealing a ref from `tok`.

File: pegen/pegen.c [hunk 1]
```diff
@@ -469,7 +469,14 @@ run_parser(struct tok_state* tok, void *(start_rule_func)(Parser *), int mode)
     }
 
     if (mode == 2) {
-        result = (PyObject *)PyAST_CompileObject(res, tok->filename, NULL, -1, p->arena);
+        PyObject *filename = (tok->filename)
```
Comment (gvanrossum) [inline pegen/pegen.c]:
```suggestion
        PyObject *filename = tok->filename
```

**Comment by gvanrossum** [comment]
Thanks!

**Pull request merged**